packages/dashboard/src/components/robots/robot-summary.tsx

```diff
@@ -0,0 +1,281 @@
+  const classes = useStyles();
+  const rmf = React.useContext(RmfAppContext);
+
+  const [isOpen, setIsOpen] = React.useState(true);
+  const [robotState, setRobotState] = React.useState<RobotState | null>(null);
+  const [taskState, setTaskState] = React.useState<TaskState | null>(null);
+  const [openTaskDetailsLogs, setOpenTaskDetailsLogs] = React.useState(false);
+  const [navigationStart, setNavigationStart] = React.useState('-');
+  const [navigationDestination, setNavigationDestination] = React.useState('-');
+
+  React.useEffect(() => {
+    if (!rmf) {
+      return;
+    }
+    const sub = rmf
+      .getFleetStateObs(robot.fleet)
+      .pipe(
+        mergeMap((fleetState) => {
+          const robotState = fleetState?.robots?.[robot.name];
+          const taskObs = robotState?.task_id ? rmf.getTaskStateObs(robotState.task_id) : of(null);
+          return robotState ? combineLatest([of(robotState), taskObs]) : EMPTY;
+        }),
+      )
+      .subscribe(([robotState, taskState]) => {
+        setRobotState(robotState);
+        setTaskState(taskState);
+      });
+    return () => sub.unsubscribe();
+  }, [rmf, robot.fleet, robot.name]);
+
+  const taskProgress = React.useMemo(() => {
+    if (
+      !taskState ||
+      !taskState.estimate_millis ||
+      !taskState.unix_millis_start_time ||
+      !taskState.unix_millis_finish_time
+    ) {
+      console.log(`Can't calculate task progress`);
+      return undefined;
+    }
+
+    return Math.min(
+      1.0 -
+        taskState.estimate_millis /
+          (taskState.unix_millis_finish_time - taskState.unix_millis_start_time),
+      1,
+    );
+  }, [taskState]);
+
+  React.useEffect(() => {
+    if (!taskState || !taskState.phases || !taskState.active) {
+      setNavigationStart('-');
+      setNavigationDestination('-');
+      return;
+    }
+
+    const message = Object.values(taskState.phases)[taskState.active - 1]?.detail;
+
+    if (message) {
+      const regex = /\[place:(.*?)\]/g;
+
+      let match;
+      const waypoints = [];
+
+      // Iterate over all matches found by the regular expression
+      while ((match = regex.exec(message.toString()))) {
+        waypoints.push(match[1]);
+      }
+
+      setNavigationStart(waypoints[0]);
+      setNavigationDestination(waypoints[1]);
+    } else {
+      setNavigationStart('-');
+      setNavigationDestination('-');
+      console.log('Failed to retrieve robot navigation');
+    }
+  }, [taskState]);
+
+  const returnDialogContent = () => {
+    const contents = [
+      {
+        title: 'Assigned tasks',
+        value: taskState ? taskState.booking.id : 'No task',
+      },
+      {
+        title: 'Est. end time',
+        value: taskState?.unix_millis_finish_time
+          ? `${new Date(taskState?.unix_millis_finish_time).toLocaleString()}`
+          : '-',
+      },
+    ];
+
+    if (taskState) {
+      contents.push(
+        {
+          title: 'Navigation start',
+          value: navigationStart,
```

Comment:
Only here will we then check if it is null, if it is, we give it `-`, probably using a ternary